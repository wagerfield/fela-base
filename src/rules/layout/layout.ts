export interface LayoutRuleOptions {
  display?: any
  direction?: any
  justify?: any
  align?: any
  wrap?: any
}

export default ({
  display = "flex",
  direction,
  justify,
  align,
  wrap
}: LayoutRuleOptions = {}) => ({
  display,
  flexWrap: wrap,
  flexDirection: direction,
  justifyContent: justify,
  alignItems: align
})
