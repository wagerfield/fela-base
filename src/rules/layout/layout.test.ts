import layout from "./layout"

describe("layout(props)", () => {
  it("renders expected snapshots", () => {
    expect(layout()).toMatchSnapshot()
    expect(
      layout({
        display: "none",
        direction: "column"
      })
    ).toMatchSnapshot()
  })
})
