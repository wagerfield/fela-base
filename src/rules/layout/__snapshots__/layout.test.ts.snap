// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout(props) renders expected snapshots 1`] = `
Object {
  "alignItems": undefined,
  "display": "flex",
  "flexDirection": undefined,
  "flexWrap": undefined,
  "justifyContent": undefined,
}
`;

exports[`layout(props) renders expected snapshots 2`] = `
Object {
  "alignItems": undefined,
  "display": "none",
  "flexDirection": "column",
  "flexWrap": undefined,
  "justifyContent": undefined,
}
`;
